File: dataset-snippets/summarize_a_survey_module.R
# Use snippet 'summarize_a_survey_module' to print a table of participant counts by question in a module
# The snippet assumes that a dataframe containing survey questions and answers already exists

# Update the next 3 lines

survey_df <- YOUR_DATASET_NAME_survey_df
module_name <- 'The Basics'
denominator <- NULL

####################################################################################
#                           DON'T CHANGE FROM HERE
####################################################################################
summarize_a_module <- function(df, module=NULL, denominator=NULL) {
    if (!is.null(module)){
        df <- df %>% filter(tolower(survey) == tolower(module))
    }
    data <- df %>% group_by(survey, question_concept_id, question) %>%
               summarize(n_participant = n_distinct(person_id))
    if (!is.null(denominator)) {
        data <- data %>% mutate(response_rate = paste0(round(100*n_participant/denominator,2),'%'))
    }
    data
}

summarize_a_module(survey_df, module_name, denominator)

File: dataset-snippets/summarize_a_survey_module.py
# Use snippet 'summarize_a_survey_module' to print a table of participant counts by question in a module
# The snippet assumes that a dataframe containing survey questions and answers already exists

# Update the next 3 lines
survey_df = YOUR_DATASET_NAME_survey_df
module_name = 'The Basics' # e.g: 'The Basics', 'Lifestyle', 'Overall Health', etc.
denominator = None # e.g: 200000

####################################################################################
#                           DON'T CHANGE FROM HERE
####################################################################################

def summarize_a_module(df, module=None, denominator=None):
    if module:
        df = df[df['survey'].str.lower() == module.lower()].copy()
    data = (df.groupby(['survey','question_concept_id','question'])['person_id'].nunique()
                .reset_index()
                .rename(columns={'person_id':'n_participant'}))
    if denominator:
        data['response_rate'] = round(100*data['n_participant']/denominator,2)
    return data

summarize_a_module(df=survey_df, module=module_name, denominator=denominator)

